unified-codes/apps/data-loader/src/app/v2/DataParser.ts

```diff
@@ -200,403 +185,83 @@ export class DataParser {
         const itemProperties: IPropertyNode[] = [];
 
         productProperties.push({
-          code: this.generateCode(),
           type: EPropertyType.RxNav,
           value: rxnav,
         });
         productProperties.push({
-          code: this.generateCode(),
           type: EPropertyType.WHOEML,
           value: who_eml_product,
         });
 
         itemProperties.push({
-          code: this.generateCode(),
           type: EPropertyType.WHOEML,
           value: who_eml_item,
         });
         productProperties.push({
-          code: this.generateCode(),
           type: EPropertyType.NZULM,
           value: nzulm,
         });
         itemProperties.push({
-          code: this.generateCode(),
           type: EPropertyType.NZULM,
           value: nzulm_item,
         });
         productProperties.push({
-          code: this.generateCode(),
           type: EPropertyType.UNSPSC,
           value: unspsc,
         });
 
-        // If row includes pack size code...
-        if (uc9) {
-          const code = uc9;
-          const name = pack_size;
-          const type = EEntityType.PackSize;
-
-          // and node does not exist...
-          if (!(uc9 in this.graph)) {
-            // create pack size node.
-            const node = {
-              code,
-              name,
-              type,
-              children: [],
-              properties: [],
-            };
-
-            this.graph[uc9] = node;
-
-            console.log(`INFO: Created node of type ${type}: ${JSON.stringify(node)}`);
-          }
-        }
-
-        // If row includes immediate packaging code...
-        if (uc8) {
-          const code = uc8;
-          const name = immediate_packaging;
-          const type = EEntityType.PackImmediate;
-
-          // and node does not exist...
-          if (!(uc8 in this.graph)) {
-            // create unit node.
-            const node = {
-              code,
-              name,
-              type,
-              children: [],
-              properties: [],
-            };
-
-            this.graph[uc8] = node;
-
-            console.log(`INFO: Created node of type ${type}: ${JSON.stringify(node)}`);
-          }
-        }
-
-        // If row includes unit of presentation code...
-        if (uc7) {
```

Comment:
uc7, 8 & 9 now have the conflict / duplicate check. that's ok?

Reply:
Hmm, I _think_ it should be ok - we don't really use/expose uc8 or uc9 at the moment - but my thinking was that they should still be unique regardless of level?

There are a few duplicates found with this update (I think maybe they should be updated in a separate issue)
- bb88f600
- 105be100
- 574f2000

I'm also a bit wary that this only checks the type...
If, say unit of use was the lowest code for two different products, and they happened to be assigned the same UC, it would probably still pass this check?

Reply:
yes - agree. I think it's fine to have the check, and we should probably be checking regardless of type. but fine for now.